File: crates/app/src/presentation/submodules.rs
use crate::style::theme::default_text_color;
use super_lazygit_core::state::SubmoduleItem;

pub fn get_submodule_list_display_strings(submodules: &[SubmoduleItem]) -> Vec<Vec<String>> {
    submodules
        .iter()
        .map(|submodule| get_submodule_display_strings(submodule))
        .collect()
}

fn get_submodule_display_strings(s: &SubmoduleItem) -> Vec<String> {
    let name = s.name.clone();
    vec![default_text_color().sprint(&name)]
}

File: pkg/gui/presentation/submodules.go
package presentation

import (
	"strings"

	"github.com/quangdang46/slg/pkg/commands/models"
	"github.com/quangdang46/slg/pkg/theme"
	"github.com/samber/lo"
)

func GetSubmoduleListDisplayStrings(submodules []*models.SubmoduleConfig) [][]string {
	return lo.Map(submodules, func(submodule *models.SubmoduleConfig, _ int) []string {
		return getSubmoduleDisplayStrings(submodule)
	})
}

func getSubmoduleDisplayStrings(s *models.SubmoduleConfig) []string {
	name := s.Name
	if s.ParentModule != nil {
		count := 0
		for p := s.ParentModule; p != nil; p = p.ParentModule {
			count++
		}
		indentation := strings.Repeat("  ", count)
		name = indentation + "- " + s.Name
	}

	return []string{theme.DefaultTextColor.Sprint(name)}
}
